### ISSUE TITLE
vagrant ssh -c , incorrect behavior

### ISSUE THREAD

**Issue by PavelPolyakov** [mitchellh/vagrant#6689]
Hi,

It seems, that when we do ```vagrant ssh -c```, the ```.profile``` is not executed.

I use `ubuntu/trusty64` machine, in my .profile I have the next part:
```
alias IP="echo $(ifconfig eth1 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}')"
```

However, when I do the next command:
```
vagrant ssh -c 'IP'
```

I receive the next:
![image](https://cloud.githubusercontent.com/assets/839290/11915670/16fafaae-a6b2-11e5-825c-718cd3df891f.png)

Even the full command leads to the same output:
![image](https://cloud.githubusercontent.com/assets/839290/11915672/28a9866c-a6b2-11e5-99fd-efe08272aeff.png)

The user is correct, so I suspect ```vagrant ssh -c``` behaviour.
I think it's not what is expected.
What do you think?

Regards,

**Comment by deancsmith** [comment]
Hi Pavel

I haven't looked at the Ruby code but my understanding of the -c option is that the command you enter is passed straight through to the guest VM. As your bash profile is on the host, then the alias does not exist on the guest?

If you were to push your alias to the .profile of your guest VM as part of the provisioning process, then this would work as expected?

**Comment by PavelPolyakov** [comment]
@deancsmith 

`vagrant ssh -c` loges in as `vagrant` user. 
`vagrant ssh` does the same.

My alias is located here `/home/vagrant/.profile`.

As you see, the users are the same:
![image](https://cloud.githubusercontent.com/assets/839290/11926340/2d915c28-a7c4-11e5-8ffd-f9f61421888c.png)

Regards,

**Comment by sethvargo** [comment]
https://github.com/mitchellh/vagrant/pull/5928 now uses the exact same command and will be released in Vagrant 1.8.

That being said, `ssh -c` and `ssh` are different commands. `ssh -c` typically does not allocate a TTY or perform a "login" which will load the .profile.
